owncloudApp/src/main/res/xml/managed_configurations.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<restrictions xmlns:android="http://schemas.android.com/apk/res/android">
+
+    <restriction
+        android:key="server_url_configuration"
+        android:restrictionType="string"
+        android:title="@string/server_url_configuration_title"
+        android:description="@string/server_url_configuration_description"
+        android:defaultValue="demo.owncloud.com" />
```

Comment:
How do default values affect the app?

For example, let's build a new OEM apk. Someone introduce in oB as server Url: `dummy.server.url`.

Let's distribute that APK without any MDM solution. Will that app show `dummy.server.url` or `demo.owncloud.com`?

Just curious 🤔

Reply:
If you distribute the app built with the MDM flavor, the MDM configurations will be prior to the brandable configurations (the ones in `setup.xml`).
If you distribute the app with the original flavor (no MDM there), the `managed_configurations.xml` will be "invisible" because no code related to retrieving data from that file will be executed, so, the brandable configuration will be the only one and prior (the previous behaviour).

Answering your question, in that case the app will show `dummy.server.url` because `demo.owncloud.com` is in the `managed_configurations.xml` file, which is irrelevant in the original flavor (the one you proposed). For the moment that's the priority policy I applied and the one I consider is more logical, but it can be changed if we decide so.

Let me know if I answered your question completely.

Reply:
My question is related to the MDM flavor. I already know that if we build with the original flavor, MDM is transparent.

But what about mdm flavor with the sample I posted?

Reply:
Ok, I get you, now code is adapted to have a more coherent behaviour when some MDM configs don't exist 😀